sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -1332,3 +1349,432 @@ class MultiFileCloudOrcPartitionReader(
+    size += 256
+    // finally the single-byte postscript length at the end of the file
+    size += 1
+    // Add in a bit of fudging in case the whole file is being consumed and
+    // our codec version isn't as efficient as the original writer's codec.
+    size + 128 * 1024
+  }
+
+  /**
+   * calculate the final block output size.
+   *
+   * There is no need to re-calculate the block size,
+   * just calculate the footer size and plus footerOffset
+   *
+   * @param footerOffset  footer offset
+   * @param blocks        blocks to be evaluated
+   * @param schema        schema info
+   * @return the output size
+   */
+  override def calculateFinalBlocksOutputSize(
+      footerOffset: Long,
+      stripes: Seq[DataBlockBase],
+      schema: SchemaBase): Long = {
+
+    // In calculateEstimatedBlocksOutputSize, we have got the true size for HEADER + All STRIPES
+    // and the estimated the FileFooter size with the worst-case. So in this case, there is no need
+    // to do re-allocate. We just ensure the final size is smaller than the initial size.
+
+    footerOffset
+  }
+
+  /**
+   * Get ThreadPoolExecutor to run the Callable.
+   *
+   * The rules:
+   * 1. same ThreadPoolExecutor for cloud and coalescing for the same file format
+   * 2. different file formats have different ThreadPoolExecutors
+   *
+   * @return ThreadPoolExecutor
+   */
+  override def getThreadPool(numThreads: Int): ThreadPoolExecutor = {
+    OrcMultiFileThreadPoolFactory.getThreadPool(getFileFormatShortName, numThreads)
+  }
+
+  /**
+   * The sub-class must implement the real file reading logic in a Callable
+   * which will be running in a thread pool
+   *
+   * @param file   file to be read
+   * @param outhmb the sliced HostMemoryBuffer to hold the blocks, and the implementation
+   *               is in charge of closing it in sub-class
+   * @param blocks blocks meta info to specify which blocks to be read
+   * @param offset used as the offset adjustment
+   * @return Callable[(Seq[DataBlockBase], Long)], which will be submitted to a
+   *         ThreadPoolExecutor, and the Callable will return a tuple result and
+   *         result._1 is block meta info with the offset adjusted
+   *         result._2 is the bytes read
+   */
+  override def getBatchRunner(
+      file: Path,
+      outhmb: HostMemoryBuffer,
+      blocks: ArrayBuffer[DataBlockBase],
+      offset: Long): Callable[(Seq[DataBlockBase], Long)] = {
+    new OrcCopyStripesRunner(file, outhmb, blocks, offset)
+  }
+
+  /**
+   * File format short name used for logging and other things to uniquely identity
+   * which file format is being used.
+   *
+   * @return the file format short name
+   */
+  override final def getFileFormatShortName: String = "ORC"
+
+  /**
+   * Sent host memory to GPU to decode
+   *
+   * @param dataBuffer  the data which can be decoded in GPU
+   * @param dataSize    data size
+   * @param isCorrectRebaseMode rebase mode
```

Comment:
would be nice to say this isn't used with ORC

Reply:
I just re-arch this function for more scalability